classes/local/v1p2/oneroster_client.php

```diff
@@ -56,5 +63,301 @@
+        try {
+            $url = new moodle_url($command->get_url($this->baseurl));
+            $params = $command->get_params();
+            $method = $command->get_method();
+            $sort = $command->get_sort();
+
+            // Add pagination.
+            if (array_key_exists('limit', $params)) {
+                $url->param('limit', $params['limit']);
+                unset($params['limit']);
+            }
+
+            if (array_key_exists('offset', $params)) {
+                $url->param('offset', $params['offset']);
+                unset($params['offset']);
+            }
+
+            // Sorting.
+            if ($sort) {
+                $url->param('sort', $sort);
+                if ($sortorder = $command->get_sort_order()) {
+                    $url->param('orderBy', $sortorder);
+                }
+            }
+
+            // Filtering.
+            if ($filter && !empty((string) $filter)) {
+                $url->param('filter', (string) $filter);
+            }
+
+            // Fields.
+            if (array_key_exists('fields', $params)) {
+                $url->param('fields', (string) $params['fields']);
+                unset($params['fields']);
+            }
+
+            if (!empty($params) && $method !== client_helper::POST) {
+                throw new BadMethodCallException(sprintf(
+                    'The http method called for %s is %s but it has to be POST' .
+                        ' if you want to pass the JSON params %s',
+                    $url,
+                    $method,
+                    json_encode($params)
+                ));
+            }
+
+            $options = [
+                'CURLOPT_CONNECTTIMEOUT' => 300,
+            ];
+
+            if ($method === client_helper::POST) {
+                $result = $this->post($url->out(false), $params, $options);
+            } else if ($method === client_helper::GET) {
+                $result = $this->get($url->out(false), $params, $options);
+            }
+
+            $info = $this->get_request_info();
+            \enrol_oneroster\local\exceptions\exception::check_and_throw_from_http_response($result, $info, $url, $params);
+
+            $response = json_decode($result);
+
+            if (is_null($response)) {
+                throw new moodle_exception("Could not decode JSON token response: " . $result);
+            }
+
+            if (!empty($response->error)) {
+                throw new moodle_exception($response->error . ' ' . $response->error_description);
+            }
+
+            // Check if response already has status info (from external API)
+            if (isset($response->imsx_statusInfo)) {
+                // Validate existing status info from external API
+                if (!$this->validateStatusInfo($response)) {
+                    throw new moodle_exception("Invalid status info found in the response");
+                }
+            } else {
```

Comment:
This should be an exception case. If there is no statusinfo, and we are in v1p2, thats not valid, and should be an exception, we shouldnt ever treat this as good